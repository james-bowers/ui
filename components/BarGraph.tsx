import * as React from "react";
import styled from "./styled";
import * as Grid from "./Grid";

export type DataPoint = {
  value: number;
};

export type BarGraphProps = {
  height: number;
  data: DataPoint[];
};

const StyledBar = styled.div`
  background: ${props => props.theme.accent};
  margin: 5px;
  align-self: flex-end;
  width: 100%;
  border-radius: 10px;
  max-width: 25px;
`;

const StyledBarContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

function StyledDataPoint(props: { dataPoint: DataPoint }) {
  return <StyledBar style={{ height: `${props.dataPoint.value}%` }} />;
}

export default function BarGraph(props: BarGraphProps) {
  return (
    <div
      style={{
        height: `${props.height}px`
      }}
    >
      <Grid.Container width="equal">
        {props.data.map((dataPoint, _index) => (
          <Grid.Item key={_index}>
            <StyledBarContainer>
              <StyledDataPoint dataPoint={dataPoint} />
            </StyledBarContainer>
          </Grid.Item>
        ))}
      </Grid.Container>
    </div>
  );
}
