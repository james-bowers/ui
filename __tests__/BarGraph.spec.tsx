import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

import BarGraph from "../components/BarGraph";

describe("<BarGraph />", () => {
  test("renders BarGraph", () => {
    const tree = shallow(
      <BarGraph
        height={200}
        data={[
          {
            value: 35
          },
          {
            value: 75
          },
          {
            value: 5
          },
          {
            value: 20
          },
          {
            value: 39
          },
          {
            value: 95
          }
        ]}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
