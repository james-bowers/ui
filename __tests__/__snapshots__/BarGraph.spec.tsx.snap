// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarGraph /> renders BarGraph 1`] = `
<div
  style={
    Object {
      "height": "200px",
    }
  }
>
  <Component
    width="equal"
  >
    <Component
      key="0"
    >
      <styled.div>
        <StyledDataPoint
          dataPoint={
            Object {
              "value": 35,
            }
          }
        />
      </styled.div>
    </Component>
    <Component
      key="1"
    >
      <styled.div>
        <StyledDataPoint
          dataPoint={
            Object {
              "value": 75,
            }
          }
        />
      </styled.div>
    </Component>
    <Component
      key="2"
    >
      <styled.div>
        <StyledDataPoint
          dataPoint={
            Object {
              "value": 5,
            }
          }
        />
      </styled.div>
    </Component>
    <Component
      key="3"
    >
      <styled.div>
        <StyledDataPoint
          dataPoint={
            Object {
              "value": 20,
            }
          }
        />
      </styled.div>
    </Component>
    <Component
      key="4"
    >
      <styled.div>
        <StyledDataPoint
          dataPoint={
            Object {
              "value": 39,
            }
          }
        />
      </styled.div>
    </Component>
    <Component
      key="5"
    >
      <styled.div>
        <StyledDataPoint
          dataPoint={
            Object {
              "value": 95,
            }
          }
        />
      </styled.div>
    </Component>
  </Component>
</div>
`;